R/commonglm.R

```diff
@@ -195,11 +195,15 @@
     l0 <- -0.5*nullModel[["deviance"]]
     lm <- as.numeric(logLik(glmModel))
     n  <- length(glmModel[["y"]])
-    coxSnell <- 1 - exp(l0 - lm)^(2 / n)
+    coxSnell <- .coxSnellCompute(l0, lm, n)
     return(max(c(0,coxSnell)))
   }
 }
 
+.coxSnellCompute <- function(l0, lm, n) {
+  return(1 - exp(2*(l0 - lm)/n))
```

Comment:
if you really want to ensure numerical accuracy this should be
```suggestion
  return(-expm1(2*(l0 - lm)/n))
```

compare for example
```r
l0 <- 2
lm <- 1.5
n <- 1e20
1 - exp(2*(l0 - lm)/n) # 0
-expm1(2*(l0 - lm)/n)  # -1e-20
```
The same goes for the other line
```
denom <- 1 - exp(2*l0/n)
```

Reply:
Great point!